src/main/kotlin/org/jetbrains/teamcity/github/util/WebHooksHelper.kt

```diff
@@ -0,0 +1,48 @@
+package org.jetbrains.teamcity.github.util
+
+import jetbrains.buildServer.serverSide.ProjectManager
+import jetbrains.buildServer.vcs.ChangesCheckingService
+import jetbrains.buildServer.vcs.OperationRequestor
+import jetbrains.buildServer.vcs.VcsRootInstance
+import org.eclipse.egit.github.core.Repository
+
+class WebHooksHelper(private val projectManager: ProjectManager, private val changesCheckingService: ChangesCheckingService) {
+
+    fun findRelevantVcsRootInstances(repository: Repository): Collection<VcsRootInstance> {
+        if (repository.cloneUrl.isNullOrEmpty())
+            return emptyList()
+
+        val repoUrls = setOf(repository.gitUrl, repository.cloneUrl, repository.sshUrl).filterNotNull().map { normalizeGitUrl(it) }
+
+        val prefilteredVcsRoots = projectManager.allVcsRoots
```

Comment:
REST API filters this collection by permission Permission.VIEW_BUILD_CONFIGURATION_SETTINGS while ProjectManager only checks VIEW_PROJECT. I think we need to use the view settings here too.

Reply:
It seems the permissions should be the same as the ones allowing to invoke "Check for pending changes" in the UI. Which seems to be RUN_BUILD. What do you think?

Reply:
Well, we need to go through settings of the entities, so view settings seems more relevant to me. Run build has own issues, I am not sure I'd want to grant run_build to an external system.

Reply:
> we need to go through settings of the entities, so view settings seems more relevant to me.

Hm, good point. 

> Run build has own issues, I am not sure I'd want to grant run_build to an external system.

This I don't get. These are permissions of the user who configured the webhook. No external build can directly run a build via this mechanism. 

Anyway, it doesn't matter much. I think the going throw settings argument has enough weight to follow your advise. Thanks!

Reply:
Checking VIEW_BUILD_CONFIGURATION_SETTINGS now. There is one minute difference to how REST operates though: for VCS root instances used in versioned settings we look at the project where the versioned settings is defined, rather to one where the VCS root is located. It makes it similar to what is done for build configurations: the projects where the build configuration belongs is checked (both by REST and the new implementation).